### PR TITLE
Remove TODO in MaxAgeCondition serialization

### LINKED ISSUE #52505: Improve rollover condition serialization
When we serialize `MaxAgeCondition` over the wire, we do `out.writeLong(value.getMillis())` while we could/should rather do `out.writeTimeValue(value)`.

Along the same lines, when we serialize `MaxSizeCondition`, we do `out.writeVLong(value.getBytes())` while we could/should rather do `value.writeTo(out)`.

Are these improvements that are worth making, also considering that they involve changing the wire format and handling backwards compatibility?

We should either make this change or alternatively remove the two corresponding TODOs in the above mentioned classes.

### PR DESCRIPTION
**Pull request by dakrone**
This removes the TODO with a message for any future readers regarding the code in question.

Resolves #52505

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MaxAgeCondition.java [hunk 1]
```diff
@@ -57,7 +57,12 @@ public String getWriteableName() {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        //TODO here we should just use TimeValue#writeTo and same for de-serialization in the constructor, we lose information this way
+        // While we technically could serialize this would out.writeTimeValue(...), that would
+        // doing the song and dance around backwards compatibility for this value. Since in this
```
Comment (jbaiera) [inline server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MaxAgeCondition.java]: "that would doing the song and dance" -> "that would require doing the song and dance"

Comment (dakrone) [reply server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MaxAgeCondition.java]: :+1: good catch

**Review by jbaiera — APPROVED** [review]
LGTM

**Comment by javanna** [comment]
heya @dakrone thanks for addressing this. There's also a TODO in MaxSizeCondition along the same lines, mentioned in the issue that this PR closed, but such TODO is still there as far as I can tell.

**Comment by dakrone** [comment]
Thanks @javanna, I'll open another PR to fix that one too.

**Pull request merged**